src/Document/Query.php

```diff
@@ -105,7 +124,7 @@ function ($matches) use ($placeholder) {
 
         foreach ($segments as $key => $segment) {
             $pathSegment = static::_tokenize($segment);
-            if (0 == $key) {
+            if (0 === $key) {
```

Comment:
Among all, this is the only change that needs attention

Reply:
Seems OK.

From psalm's types:

https://github.com/vimeo/psalm/blob/ab5ddb1514f7de0976b964e92e9bf51138aa1496/stubs/CoreGenericFunctions.phpstub#L593-L606

Seems like `preg_split` returns `non-empty-list<string>|false` when `$flags = 0` (default case)